corehq/apps/userreports/tests/test_choice_provider.py

```diff
@@ -506,3 +508,74 @@ def test_get_choices_for_values(self):
             ['unknown-user'] + [user._id for user in self.users],
             self.web_user,
         )
+
+
+class DomainChoiceProviderTest(TestCase, ChoiceProviderTestMixin):
+    domain = "domain-choicer-provider"
+
+    @classmethod
+    def setUpClass(cls):
+        super(DomainChoiceProviderTest, cls).setUpClass()
+        report = ReportConfiguration(domain="A")
+
+        invitations = [Invitation('A'), Invitation('B'), Invitation('C')]
+        # A, B, and C are in the registry A has access to B and C, B has access to C
+        grants = [
+            Grant("C", ["B", "A"]),
+            Grant("B", ["A"]),
+            Grant("A", []),
+        ]
+        cls.registry = create_registry_for_test(cls.domain, invitations, grants, name="registry")
+
+        choices = [
+            SearchableChoice("A", "A", ["A"]),
+            SearchableChoice("B", "B", ["B"]),
+            SearchableChoice("C", "C", ["C"]),
+        ]
+        choices.sort(key=lambda choice: choice.display)
+        cls.choice_provider = DomainChoiceProvider(report, None)
+        cls.static_choice_provider = StaticChoiceProvider(choices)
+
+        cls.web_user = UserChoiceProviderTest.make_web_user('web-user@example.com', domain="A")
+
+    @classmethod
+    def tearDownClass(cls):
+        delete_all_users()
+        super(DomainChoiceProviderTest, cls).tearDownClass()
+
+    def test_query_search(self):
+        self._test_query(ChoiceQueryContext("A", limit=1, page=0))
+
+    def test_query(self):
+        self.assertEqual(1, len(self.choice_provider.query(ChoiceQueryContext(query='A', offset=0))))
+        self.assertEqual(3, len(self.choice_provider.query(ChoiceQueryContext(query='', offset=0))))
+        self.assertEqual(0, len(self.choice_provider.query(ChoiceQueryContext(query='D', offset=0))))
```

Comment:
by checking counts here you're effectively doing the same as `test_query_count`. It would be better to check the actual values.